xstate-chatbot/nodejs/src/machine/hospital-details.js

```diff
@@ -464,25 +509,37 @@ const hospitalFlow = {
         },
         process: {
           onEntry: assign((context, event) => {
-            if (event.message.type == 'text' && context.slots.previoushospitaldata.hospital_level_Id == '2') {
-              const message = dialog.get_input(event, false);
-              context.slots.hospital.bed_vacant_L2 = message;
-              context.validMessage = true;
-            } else
-              if (event.message.type == 'text' && context.slots.previoushospitaldata.hospital_level_Id == '1') {
-                const message = dialog.get_input(event, false);
+            const message = dialog.get_input(event, false);
+            const availableBed = parseInt(message);
+            const totalBedCount = parseInt(context.slots.previoushospitaldata.bed_capacity_L2);
+
+            if (event.message.type == 'text' && availableBed > totalBedCount)
+              context.isValid = true;
+            else
+            if (event.message.type == 'text' && context.slots.previoushospitaldata.hospital_level_Id == '2' && availableBed <= totalBedCount) {
+               context.slots.hospital.bed_vacant_L2 = message;
+               context.isValid = false;
+               context.validMessage = true;
+             } else
+            if (event.message.type == 'text' && context.slots.previoushospitaldata.hospital_level_Id == '1' && availableBed <= totalBedCount) {
                 context.validMessage = false;
+                context.isValid = false;
                 context.slots.hospital.bed_vacant_L2 = message;
-              }
+                }
           }),
           always: [
+            {
+              cond: (context) => context.isValid == true,
```

Comment:
If it is valid, then also you want to go to error state?
If you are using the .isValid variable to store something other than validity of the message, then I suggest use some other variable name.

Reply:
While entering  requested data (message value that is no of available of beds)  ,we are comparing this with the total beds ...And if it is greater than Total Beds then again requested to nodal officers to re-enter the correct (less than total ) beds number.

Reply:
Okay. But, I am suggesting use some other meaningful name for that variable instead of "isValid". For any new developer it would be confusing that why are we going to error state even when the input is valid.

Reply:
Changed variable and committed code